samples/scheduling/hangfire/Core_6/Scheduler/Program.cs

```diff
@@ -0,0 +1,62 @@
+﻿using System;
+using System.Threading.Tasks;
+using Hangfire;
+using Hangfire.MemoryStorage;
+using NServiceBus;
+using NServiceBus.Logging;
+using NServiceBus.Serilog;
+using Scheduler;
+using Serilog;
+
+class Program
+{
+    static async Task Main()
+    {
+        #region serilog
+
+        Log.Logger = new LoggerConfiguration()
+            .WriteTo.ColoredConsole()
+            .CreateLogger();
+        LogManager.Use<SerilogFactory>();
+
+        #endregion
+
+        Console.Title = "Samples.HangfireScheduler.Scheduler";
+        var endpointConfiguration = new EndpointConfiguration("Samples.HangfireScheduler.Scheduler");
+        endpointConfiguration.UseTransport<LearningTransport>();
+
+        #region Configuration
+
+        var endpointInstance = await Endpoint.Start(endpointConfiguration)
+            .ConfigureAwait(false);
+
+        // store the endpointInstance in a static helper class
+        EndpointHelper.Instance = endpointInstance;
+
+        //use in memory storage. Production scenarios should use more robust alternatives: SqlServer, Msmq, Redis etc
+        GlobalConfiguration.Configuration.UseMemoryStorage();
+
+        // create and start scheduler instance
+        var scheduler = new BackgroundJobServer();
+
+        #endregion
+
+        #region scheduleJob
+        
+        // Tell Hangfire to schedule the job and trigger every minute 
+        RecurringJob.AddOrUpdate("myJob", () => SendMessageJob.Run(), Cron.Minutely);
```

Comment:
Why not change to `RecurringJob.AddOrUpdate("myJob", () => SendMessageJob.Run(endpointInstance), Cron.Minutely);` and then 

```
public class SendMessageJob
    {
        public static Task Run(IMessageSession session)
        {
            return session.Send("Samples.HangfireScheduler.Receiver", new MyMessage());
        }
    }
```

Reply:
That will not work, unfortunately. Hangfire serialized the expression and the arguments into a JSON blob, which stored until the trigger and then deserialized and run.

Reply:
Which can run at a different machine if you run multiple instances etc